lib/Smokeping/Master.pm

```diff
@@ -123,7 +123,7 @@ sub save_updates {
         }
     }
     for my $name (sort keys %u){
-        my $file = slavedatadir($cfg) ."/${name}.${slave}.slave_cache";
+        my $file = slavedatadir($cfg) ."${name}.${slave}.slave_cache";
```

Comment:
slavedatadir never ends in a slash ... so dropping the slash here has drastic consequences ...